commit 4b76ac0bff27c0a32fc22591ec4543768ae5b991
Author: Anthony G. Basile <blueness@gentoo.org>
Date:   Sat Apr 5 16:32:30 2014 -0400

    Revisit issue #87: fails to build on linux headers 3.13
    
    We switch to defining _GNU_SOURCE as we do for cdrom_id.c
    for a more consistant approach to this issue.
    
    Signed-off-by: Anthony G. Basile <blueness@gentoo.org>

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff --git a/src/accelerometer/accelerometer.c b/src/accelerometer/accelerometer.c
index beed53c..501caa4 100644
--- a/src/accelerometer/accelerometer.c
+++ b/src/accelerometer/accelerometer.c
@@ -44,6 +44,10 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE 1
+#endif
+
 #include <stdio.h>
 #include <string.h>
 #include <stdbool.h>
@@ -58,10 +62,6 @@
 #include <linux/limits.h>
 #include <linux/input.h>
 
-#ifdef __UCLIBC__
-#include <linux/fcntl.h>
-#endif
-
 #include "libudev.h"
 #include "libudev-private.h"
 
diff --git a/src/ata_id/ata_id.c b/src/ata_id/ata_id.c
index 93dbb53..b416cde 100644
--- a/src/ata_id/ata_id.c
+++ b/src/ata_id/ata_id.c
@@ -19,6 +19,10 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE 1
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
@@ -41,10 +45,6 @@
 #include <linux/bsg.h>
 #include <arpa/inet.h>
 
-#ifdef __UCLIBC__
-#include <linux/fcntl.h>
-#endif
-
 #include "libudev.h"
 #include "libudev-private.h"
 #include "log.h"
diff --git a/src/collect/collect.c b/src/collect/collect.c
index 9d1a97e..05d6517 100644
--- a/src/collect/collect.c
+++ b/src/collect/collect.c
@@ -19,6 +19,10 @@
  *
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE 1
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stddef.h>
@@ -31,10 +35,6 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
-#ifdef __UCLIBC__
-#include <linux/fcntl.h>
-#endif
-
 #include "libudev.h"
 #include "libudev-private.h"
 #include "macro.h"
diff --git a/src/mtd_probe/mtd_probe.c b/src/mtd_probe/mtd_probe.c
index ffba0eb..0fbf3ab 100644
--- a/src/mtd_probe/mtd_probe.c
+++ b/src/mtd_probe/mtd_probe.c
@@ -16,6 +16,11 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA  02110-1301  USA
  */
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE 1
+#endif
+
 #include "mtd_probe.h"
 #include <stdio.h>
 #include <sys/ioctl.h>
@@ -26,10 +31,6 @@
 #include <unistd.h>
 #include <stdlib.h>
 
-#ifdef __UCLIBC__
-#include <linux/fcntl.h>
-#endif
-
 int main(int argc, char** argv)
 {
         int mtd_fd;
diff --git a/src/scsi_id/scsi_serial.c b/src/scsi_id/scsi_serial.c
index b9a94e6..c3f957d 100644
--- a/src/scsi_id/scsi_serial.c
+++ b/src/scsi_id/scsi_serial.c
@@ -17,6 +17,10 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE 1
+#endif
+
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
@@ -34,10 +38,6 @@
 #include <linux/types.h>
 #include <linux/bsg.h>
 
-#ifdef __UCLIBC__
-#include <linux/fcntl.h>
-#endif
-
 #include "libudev.h"
 #include "libudev-private.h"
 #include "scsi.h"
